spec/models/location_spec.rb

```diff
@@ -930,6 +930,29 @@
     end
   end
 
+  describe "status_at" do
+    let(:location) { FactoryBot.build(:location, startdate: Time.zone.local(2022, 4, 1)) }
+
+    before do
+      Timecop.freeze(2022, 6, 7)
+    end
+
+    after do
+      Timecop.unfreeze
+    end
+
+    context "when there have been previous deactivations" do
+      before do
+        FactoryBot.create(:location_deactivation_period, deactivation_date: Time.zone.local(2022, 5, 4), reactivation_date: Time.zone.local(2022, 6, 5), location:)
+        location.save!
+      end
+
+      it "returns active if the location has no relevant deactivation records" do
+        expect(location.status_at(Time.zone.local(2022, 4, 4))).to eq(:active)
+      end
+    end
```

Comment:
Is it worth adding in test for when this should return other statuses too?

Reply:
I've added them initially, but they're all sort of implicitly being tested by
```
  def status
    @status ||= status_at(Time.zone.now)
  end
```

Reply:
Hmm, yeah good point. All looks good then!